source/external/Box2D/CMakeLists.txt

```diff
@@ -56,3 +60,10 @@ if (NOT IOS)
 endif()
 
 target_include_directories(Box2D PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
+
+set_target_properties(Box2D
```

Comment:
Please remove set_target_properties. (at least for know) 
I propose to use for know only:
```
target_include_directories
target_link_libraries
add_subdirectory
add_library
```

Reply:
I don't think this is a good idea. for example, properties `FOLDER "External"`, move the long list library target into `External` target when generated VS project.

if we need removed it now, and re-add it later. why not leave it here.

Reply:
OK - for FOLDER.
But please remove ARCHIVE_OUTPUT_DIRECTORY, LIBRARY_OUTPUT_DIRECTORY.

https://cmake.org/cmake/help/latest/prop_tgt/ARCHIVE_OUTPUT_DIRECTORY.html
https://cmake.org/cmake/help/v3.12/prop_tgt/LIBRARY_OUTPUT_DIRECTORY.html
`
This property specifies the directory into which library target files should be built. The property value may use generator expressions. Multi-configuration generators (VS, Xcode) append a per-configuration subdirectory to the specified directory unless a generator expression is used.`

I thin that setting ARCHIVE_OUTPUT_DIRECTORY, LIBRARY_OUTPUT_DIRECTORY prevent VC, Xcode to have multiple configs available e.g.: Release, Debug

Reply:
think again, I removed the `set_target_properties`, just keep simple and effective.